src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (tail == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size() || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds.");
+        }
+        if (index == size) {
+            add(value);
```

Comment:
you have method uncurrentIndex, maybe it would be better to use it here instead of these checks

Reply:
In method uncurrentIndex - if (index >= size) , but i need index > size
That's why i can't use this method there